python/ray/air/result.py

```diff
@@ -48,6 +58,7 @@ class Result:
     _local_path: Optional[str] = None
     _remote_path: Optional[str] = None
     _items_to_repr = ["error", "metrics", "path", "checkpoint"]
+    _restore_required_files = [EXPR_RESULT_FILE]
```

Comment:
Oh, for this, I meant:

- We shouldn't raise a warning if the user doesn't have *both* files. They could have run with only the JSON logger or only the CSV logger. The user must run with the JSON logger for the current code to work.
- What do you think about:
    - Fallback to CSV if JSON not found.
    - Just keep it as is and require JSON, I think that's fine since 99% of the time users will run with both (that's the default). But, we should plan when to get rid of csv logger.

Reply:
Oh got it. In that case, I prefer to fallback to loading CSV if there is no JSON for robustness. Then raise a deprecation warning if they are using csv logger.